### PR TITLE
Feature/17976 complete check data service

### PR DESCRIPTION
**Pull request by GuyHarwood**
Incorporates work on marking service, ps report service and check service.

### CODE REVIEW

File: admin/spec/service/marking.service.spec.js [hunk 1]
```diff
@@ -1,34 +1,21 @@
...
 const completedCheckDataService = require('../../services/data-access/completed-check.data.service')
 const completedCheckMock = require('../mocks/completed-check')
 
 describe('markingService', () => {
-  let service, sandbox
+  let service = require('../../services/marking.service')
 
   beforeEach(() => {
-    sandbox = sinon.sandbox.create()
+    // sandbox = sinon.sandbox.create()
```
Comment (GeorgeChatzigiannis) [inline admin/spec/service/marking.service.spec.js]: I am guessing you can just delete the sandbox creation and restoration right?

File: admin/services/check-complete.service.js [hunk 2]
```diff
@@ -23,8 +23,8 @@ checkCompleteService.completeCheck = async function (completedCheck) {
   completedCheck.receivedByServerAt = moment.utc()
 
   // store to data store
-  await completedChecksDataService.create(completedCheck)
-  // temporary way to mark checks until we move to a dedicated scheduled process
+  await completedCheckDataService.sqlAddResult(completedCheck)
+  // HACK temporary way to m ark checks until we move to a dedicated scheduled process
```
Comment (jon-shipley) [inline admin/services/check-complete.service.js]: unwanted space in 'mark'

File: admin/spec/service/marking.service.spec.js (hunk 1 above)
Comment (GuyHarwood) [reply admin/spec/service/marking.service.spec.js]: yep, probably just forgot to remove it once i got the tests passing

File: admin/services/data-access/completed-check.data.service.js [hunk 3]
```diff
@@ -80,14 +154,37 @@ completedCheckDataService.findUnmarked = async function (batchSize) {
   return batchIds.map(b => b._id)
 }
 
+/**
+ * Return plain completedCheck objects that have not been marked, up to a the limit specified by `batchsize`
```
Comment (jon-shipley) [inline admin/services/data-access/completed-check.data.service.js]: I think the documentation and the code are out of step.  This returns an array of numbers (id's) not objects.

File: admin/services/data-access/completed-check.data.service.js [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 
 const CompletedChecks = require('../../models/completed-checks')
+const sqlService = require('./sql.service')
+const TYPES = require('tedious').TYPES
+const R = require('ramda')
 const completedCheckDataService = {}
 
 /**
  * Query : find a completedCheck document that has not been marked
+ * @deprecated
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/completed-check.data.service.js]: maybe just for consistency just add something like: @deprecated unused - no replacement

**Pull request merged**